rust/src/datatypes.rs

```diff
@@ -338,6 +340,29 @@ impl Field {
                     panic!("Unsupported dictionary key type: {}", key_type);
                 }
             },
+            DataType::Struct(mut subfields) => {
+                for (i, f) in subfields.iter_mut().enumerate() {
+                    let lance_field = self
+                        .children
+                        .iter_mut()
+                        .find(|c| c.name == f.name().to_string())
+                        .unwrap();
+                    let struct_arr = arr.as_any().downcast_ref::<StructArray>().unwrap();
+                    lance_field.set_dictionary(struct_arr.column(i));
+                }
+            }
+            DataType::List(values) => {
+                // should this be self.children[0]?
+                let mut f = Field::try_from(values.as_ref()).unwrap();
```

Comment:
change to use self.children[0]?

I might prefer keep such case unhandled instead of leaving code like this.